docs/upgrading.md

```diff
@@ -0,0 +1,28 @@
+# Upgrading
+
+## Upgrading from v2 to v3.0.0
+
+Starting from `external-dns` version `3.0.0` and later, there have been modifications to the values.yaml schema. These changes were introduced in version `2.28.0`, and the previous format has been completely removed in version 3. Therefore, it's important to take note of these updates while upgrading.
```

Comment:
What's the official name then? I see the chart directory is still called `external-dns-app`. Wouldn't this be the perfect moment to rename it to just `external-dns`?

Reply:
I'll run some tests to see if it's feasible.

Reply:
I ran a test and it actually breaks vintage. I'm no sure if I really want to go that road. On the other hand, that forces a change in the deployments selectorLabels, which was left out of the alignment. 

In any case. this main-v3 branch is most of the work for alignment but not all. I don't expect to release immediately and any extra changes can be added in a follow-up PR.